src/Symfony/Component/HttpFoundation/HeaderBag.php

```diff
@@ -121,7 +121,15 @@ public function get($key, $default = null, $first = true)
         }
 
         if ($first) {
-            return \count($headers[$key]) ? (string) $headers[$key][0] : $default;
+            if (!$headers[$key]) {
```

Comment:
When does this happen?

Reply:
Check the tests: https://github.com/symfony/symfony/pull/33353/files#diff-1f86cfdac587d9d9aca829f2b713c463R374

This happens to us when generating a response in a bit more complicated way. I think that is not what should happen in this case.

Reply:
But in that case `!$headers[$key]` is `[null]`, isn't it? And the following condition (lines 128 to 130) will be hit.

Reply:
Ah, sry, my bad. This happens when the `$headers[$key]` is empty... The logic was the same before (just written as a ternary), and I don't know when that happens :man_shrugging: